File: Java/Laba8/src/main/java/BookShop.java
import java.util.ArrayList;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.List;

public class BookShop {
    public List<Good> goodsList = new LinkedList<>();

    public final List<Good> getGoodsList() {
        return goodsList;
    }

    public BookShop(List<Good> goods) {
        this.goodsList = goods;
    }

    public List<Good> getGoods() {
        return goodsList;
    }

    public void setGoods(List<Good> goods) {
        this.goodsList = goods;
    }

    /**
     * Add decoration to decoration's list
     *
     * @param goods object of the parent class of all decorations

    public final void addGood(final Good decoration) {
        this.goods.add(goods);
    }*/

    public void sortByPages(List<Good> goods) {
        goods.sort(Comparator.comparingInt(Good::getPages));
    }

    public List<Good> searchByCategory(CategoryEnum categoryEnum) {
        List<Good> listInput = new ArrayList<Good>();
        for (int i = 0; i < goodsList.size(); i++) {
            if (goodsList.get(i).categoryEnum == categoryEnum) {
                listInput.add(goodsList.get(i));
            }
        }
        return listInput;
    }
}

File: Python/lab14/BookShop.py
class BookShop:
    good_list = []

    def __init__(self):
        pass

    def sort_by_pages(self):
        self.good_list.sort(key=lambda goods: goods.pages)
        return self.good_list

    def find_by_category(self, category):
        found_good = []
        for good in self.good_list:
            if good.category == category:
                found_good.append(good)
        return found_good

    def print_list(self):
        for good in self.good_list:
            print(good)
        print("\n")
